x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/input/FrozenIndexInput.java

```diff
@@ -255,28 +239,19 @@ private void writeCacheFile(
             cacheFile
         );
         final long end = relativePos + length;
-        final byte[] copyBuffer = new byte[toIntBytes(Math.min(COPY_BUFFER_SIZE, length))];
         logger.trace(() -> format("writing range [%s-%s] to cache file [%s]", relativePos, end, cacheFile));
 
         long bytesCopied = 0L;
         long remaining = length;
-        final ByteBuffer buf = writeBuffer.get();
-        buf.clear();
+        final ByteBuffer buf = writeBuffer.get().clear();
```

Comment:
Just a nit but I wonder if we could make `writeBuffer` private and add a protected `getWriteBuffer()` in `MetadataCachingIndexInput` that ensures the `ByteBuffer` is properly cleared before usage? We could also check which threads are accessing it.

Reply:
I'm always a little hesitant hiding the fact that something is thread-local behind another layer to be honest. I agree it looks nicer here, but it makes it much easier to take a wrong step for someone new to this code here.

Reply:
Yeah, that's a valid argument too.